tasks/solve-a-hidato-puzzle/src/main.rs

```diff
@@ -0,0 +1,106 @@
+use std::cmp::{max, min};
+
+#[derive(Clone, Debug)]
+struct Puzzle {
+    board: Vec<Vec<i32>>,
+    fixed: Vec<i32>,
```

Comment:
Can you add a doc comments here? What is `fixed`?

Reply:
Done as well.